daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/PrimitivesTextNumber1.scala

```diff
@@ -54,11 +56,58 @@ case class ConvertTextCombinatorParser(
   }
 }
 
+trait TextDecimalVirtualPointMixin {
+  def textDecimalVirtualPoint: Int
+
+  final protected val divisor = scala.math.pow(10.0, textDecimalVirtualPoint)
+  final protected val multiplier = scala.math.pow(10.0, textDecimalVirtualPoint)
```

Comment:
I'm confused, how come you are defining both divisor and multiplier as the same value?  Please add a comment explaining why.

Reply:
Ha! it's really the "magic number" and one code path multiplies by it. the other divides by it. I'll explain.